### PR TITLE
diann 1.8.1 -> 1.9.1dev

### PR DESCRIPTION
**Pull request by daichengxin**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/quantms/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/quantms _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -65,9 +68,34 @@ jobs:
           echo "$(pwd)/micromamba/bin" >> $GITHUB_PATH
           ./bin/micromamba shell init -s bash
           echo $'channels:\n  - conda-forge\n  - bioconda\n  - defaults\nuse_lockfiles: false' >> ~/.mambarc
-
-      - name: Run pipeline with test data
-        if: matrix.exec_profile != 'conda'
+      - name: Install Singularity
```
Comment (jpfeuffer) [inline .github/workflows/ci.yml:71]: There is an action for this:
https://github.com/singularityhub/install-singularity

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -70,31 +70,11 @@ jobs:
...
-          sudo tar -C /usr/local -xzf go1.22.5.linux-amd64.tar.gz
-          echo 'export PATH=/usr/local/go/bin:$PATH' >> ~/.bashrc && \
-          source ~/.bashrc
-          wget -qO- https://github.com/sylabs/singularity/releases/download/v4.1.4/singularity-ce_4.1.4-jammy_amd64.deb
-          sudo apt install ./singularity-ce_4.1.4-jammy_amd64.deb
+        steps:
+          - name: Checkout Repository
+            uses: actions/checkout@v3
+          - name: Singularity install with defaults
+            uses: singularityhub/install-singularity@main
```
Comment (jpfeuffer) [inline .github/workflows/ci.yml:77]:
```suggestion
        - name: Singularity install with defaults
          uses: singularityhub/install-singularity@main
```

**Review by ypriverol — APPROVED** [review]

**Review by ypriverol — APPROVED** [review]

**Review by jpfeuffer — APPROVED** [review]

**Comment by wfondrie** [comment]
Was there an update to the DIANN license that change the conclusion here (https://github.com/bigbio/quantms/pull/381#issuecomment-2208585351)?
> We are closing this PR. In the current LICENSE of DIANN, cloud infrastructures are not allowed.

**Comment by wfondrie** [comment]
@ypriverol - this is the contents of the license distributed with https://github.com/vdemichev/DiaNN/releases/tag/1.9.1:

>  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
> INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. 
> IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, 
> OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, 
> OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, 
> OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, 
> EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
> 
> Herein the terms 'DIA-NN' or 'DIA-NN software' are used to refer to specifically the
> version of the DIA-NN software which packs this license file in its distribution package.
> 
> 1. Special DIA-NN software usage scenarious.
> Collaborative use refers to any usage of the DIA-NN software solely for the purpose of
> carrying out a joint collaborative project between yourself or your insititution and the Demichev 
> laboratory at the Charité – Universitätsmedizin Berlin. The nature of the project and the intended 
> use of the DIA-NN software in this case must be agreed with the Demichev laboratory in advance.
> 
> 2. The DIA-NN software is licensed to you for use only under the terms of this License, 
> and all rights not expressly granted to you are reserved. The rights granted herein are limited to Vadim Demichev’s intellectual
> property rights in the DIA-NN software and do not include any other patents or intellectual property rights.
> 
> 3. Permitted use. Provided the usage restrictions imposed by this License are satisfied,
> this License allows you to install and use the DIA-NN software either (i) solely for the purpose of
> academic research or education or (ii) under the Collaborative use usage scenario or 
> (iii) solely for the purpose of publishing a benchmark that either demonstrates the performance 
> of a particular method / technology / data analysis approach or demonstrates the qualities of a particular data set.
> You may make one copy of the DIA-NN software in machine-readable form for backup purposes only, 
> provided that the backup copy must include all copyright or other proprietary notices contained on the original.
> Usage of DIA-NN in the cloud is only permitted under Collaborative use. Except as and only to the extent expressly 
> permitted in this License or by applicable law, you may not copy, decompile, reverse engineer, disassemble, 
> attempt to derive the source code of, modify, or create derivative works of the DIA-NN software or any part thereof. 
> 
> 4. Transfer. You may not rent, lease, lend or sublicense the DIA-NN software. You may, however, make a one-time permanent transfer 
> of all of your license rights to the DIA-NN software to another party, provided that: (a) the transfer must include all of the DIA-NN software, 
> including all its component parts, original media, printed materials and this License; (b) you do not retain any copies of the DIA-NN software, 
> full or partial, including copies stored on a computer or other storage device; and (c) the party receiving the DIA-NN software reads 
> and agrees to accept the terms and conditions of this License.
> 
> 5. Attribution. If you share, distribute or publish any materials that were produced using the DIA-NN software ("Derived Materials"), you must indicate 
> within these Derived Materials that they were produced using the DIA-NN software. You may satisfy this condition in any reasonable manner 
> based on the medium, means, and context in which you share the Derived Materials. For example, the required attribution can be expressed by including
> one of the statements "Processed using DIA-NN", "Generated using DIA-NN" or "Produced using DIA-NN" in the Derived Materials.
> 
> 6. All third-party software included in this DIA-NN software distribution retains its original license.
> For the inclusion in this DIA-NN software distribution, the source code of MSToolkit and MiniDNN was modified.
> If not packaged within this distribution, please find the source code of the third-party software at https://github.com/vdemichev/DiaNN
> or at the respective website of the third-party software.
> 
> 7. Experimental functionality.
> Any of the DIA-NN commands/options that are not referenced at https://github.com/vdemichev/DiaNN/blob/master/README.md 
> are considered Experimental functionality. The following DIA-NN commands/options are also considered Experimental functionality:
> --extract
> --mgf
> Experimental functionality is in the process of being actively developed and is thus shared 'on collaborative basis' 
> before it is published in a peer-reviewed publication. It is not tested as thoroughly as the rest of DIA-NN and 
> needs to be used with caution, with expertise in the internal workings of DIA-NN potentially required for correct use. 
> Any peer-reviewed publication or academic preprint derived from any usage of Experimental functionality that occurs 
> before a peer-reviewed manuscript describing the respective Experimental functionality is published must therefore 
> be published in collaboration and co-authorship with the Demichev laboratory. Using Experimental functionality is not
> permitted if you do not agree to the above. A peer-reviewed manuscript is considered to describe a particular 
> Experimental functionality if (i) it is referenced at https://github.com/vdemichev/DiaNN/blob/master/README.md 
> as describing this functionality and (ii) the date when it was made available online by the publisher falls after 
> the date on which the specific version of the DIA-NN software to which this license refers to was built.
> 

A couple key provisions will prevent us from using quantms going forward unless the version is reverted:
1. Commercial use is not granted:
> this License allows you to install and use the DIA-NN software either (i) solely for the purpose of
academic research or education or (ii) under the Collaborative use usage scenario or 
(iii) solely for the purpose of publishing a benchmark that either demonstrates the performance 
of a particular method / technology / data analysis approach or demonstrates the qualities of a particular data set.

2. Only collaborators can run DIANN in the cloud:
> Usage of DIA-NN in the cloud is only permitted under Collaborative use.

**Comment by ypriverol** [comment]
I discussed with @vdemichev and the license explicitly said that cloud use (for me meaning docker in the context of quantms) is not allowed even in an academic context. Then, we have decided to support singularity. 

Probably  @vdemichev can make everything more clear in the issue. Singularity will allow the users to run it locally, and also in HPC infrastructure.

**Comment by wfondrie** [comment]
Thanks @ypriverol! Sorry I was quick to respond before diving into the contents of this PR.

**Comment by ypriverol** [comment]
It is fine. This topic is complex because also docker can be used locally but the main advantage is to use it in the cloud and because the cloud is now "not permitted", we can't be enablers of it.

**Comment by jpfeuffer** [comment]
Yes we explicitly decided to default to 1.8.1 due to the license, right @ypriverol ?

For 1.8.1 we support the biocontainer images for docker and singularity.

But people that want to try the new version can do so now with a custom singularity image.
Personally I would have been fine with just fixing the parsing scripts and letting the user  overwrite the config everytime but here we introduce a parameter that automatically picks the right image etc.

By the way @ypriverol We should probably print an error if the "backend" is AWS/AWS batch with this version.

**Pull request merged**